pymc_marketing/mmm/base.py

```diff
@@ -343,19 +350,28 @@ def plot_posterior_predictive(
             )
 
             ax.fill_between(
-                x=self.X[self.date_column],
+                x=posterior_predictive_data.date,
                 y1=likelihood_hdi_50[:, 0],
                 y2=likelihood_hdi_50[:, 1],
                 color="C0",
                 alpha=0.3,
                 label="$50\%$ HDI",  # noqa: W605
             )
 
-            target_to_plot: np.ndarray = np.asarray(
-                self.y if original_scale else self.preprocessed_data["y"]  # type: ignore
+            target_to_plot = np.asarray(
+                self.y
+                if original_scale
+                else self.get_target_transformer().transform(self.y[:, None]).flatten()  # type: ignore
             )
+
+            assert len(target_to_plot) == len(posterior_predictive_data.date), (
```

Comment:
Can you use an if statement and raise an expectation instead? The assert statements can bring weird consequences, https://bandit.readthedocs.io/en/latest/plugins/b101_assert_used.html

Reply:
can we wrap `self.get_target_transformer().transform(self.y[:, None]).flatten()` into a little function? (and add a test to document what we are doing?)

Reply:
> Can you use an if statement and raise an expectation instead? The assert statements can bring weird consequences, https://bandit.readthedocs.io/en/latest/plugins/b101_assert_used.html
✅

> can we wrap self.get_target_transformer().transform(self.y[:, None]).flatten() into a little function? (and add a test to document what we are doing?)

I see, but not sure I agree. It's a simple expression (y a vertical vector, because that's what the transformer expects, then transform it into model scale and flatten it to an 1d array). @wd60622 thoughts?

Reply:
I am for making a small function. [This function](https://github.com/pymc-labs/pymc-marketing/blob/582d2ed4ddaa464cfe48dd52103afd5cb9001a4b/pymc_marketing/mmm/utils.py#L242) needs a friend 🤗
Something like `transform_1d_array`

Reply:
Will make a util then :)